scripts/filter_multimappers.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python
+
+import argparse
+import pysam
+import sys
+
+def parse_arguments():
+    '''
+    Command-line arguments parser
+    '''
+    parser = argparse.ArgumentParser(
+        description="Script to extend pre-miRNs overhang."
+        )
+
+    parser.add_argument(
+        '-v','--version', 
+        action='version', 
+        version='%(prog)s 1.0',
+        help="Show program's version number and exit"
+        )
+
+    parser.add_argument(
+        '-i', '--input', 
+        help="Path to the gff3 input file",
+        type=str,
+        default=None,
+        )
+
+    parser.add_argument(
+        '-o', '--output', 
+        help="Path for the gff3 final file.",
+        type=str,
+        default=None,
+        )
+    return parser
+
+
+def count_matches(read):
+    '''
+    Counts the number of matches in a read based on its CIGAR string.
+
+    Args:
+        read (pysam.AlignedSegment): 
+            The read to count matches for.
+
+    Returns:
+        int: The number of matches in the read.
+    '''
+    
+    return sum([op[1] for op in read.cigar if op[0] == 0])
+
+def count_indels(read):
+    '''
+    Counts the number of insertions and/or deletions in a read based on its 
+    CIGAR string.
+
+    Args:
+        read (pysam.AlignedSegment): The read to count insertions for.
+
+    Returns:
+        int: The number of insertions in the read.
+    '''
+    return sum([op[1] for op in read.cigar if op[0] == 1 or op[0] == 2])
+
+def filter_multimappers(in_sam_file, out_sam_file):
+    '''
+    Filters multimappers from a SAM file and writes the filtered reads to a new
+    SAM file.
+
+    This function scans a SAM file recording how many alignments per read there
+    are. Then, for the reads with multiple alignments (multimappers), it counts
+    the number of matches and indels it contain. Finally, it writes to a new SAM
+    file the alignments with the more matches and the less indels. If different 
+    alignments obtain the same score, all of them will be kept.
```

Comment:
I think you can move this to the top as a module level docstring (see comments above and below). However, I don't think it's very well written. Perhaps you can ask ChatGPT to explain your script to you and then use that (following the suggested summary line from above) :)